### PR TITLE
Basic integration

### PR DESCRIPTION
**Pull request by voisardf**
* I integrated most of the project I started seperately
* I added some checks to the make file and flaked the project
For this one I didn't fork the project yet, but I will just after merging this PR :)

Please give it a look. Cheers! :)

### CODE REVIEW

File: sitn_portal/sitn-portal-client/src/app/docs/document.service.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  constructor(private http: Http) { }
+
+  getDocuments(): Promise<Document[]> {
+    return this.http.get(this.documentsUrl)
+       .toPromise()
+       .then(response => response.json().documents as Document[])
+       .catch(this.handleError);
+     }
+
+  private headers = new Headers({'Content-Type': 'application/json'});
```
Comment (kalbermattenm) [inline sitn_portal/sitn-portal-client/src/app/docs/document.service.ts]: Is this (still) necessary ?

File: sitn_portal/sitn-portal-client/src/app/docs/document.service.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export class DocumentService {
+  getDocument(docid: string): Promise<Document> {
+    const url = `${this.documentsUrl}/${docid}`;
+    return this.http.get(url)
+      .toPromise()
+      .then(response => response.json().documents as Document)
+      .catch(this.handleError);
+  }
+
+  private documentsUrl = sitnDocsConst['documentsUrl'];
```
Comment (kalbermattenm) [inline sitn_portal/sitn-portal-client/src/app/docs/document.service.ts]: I would reorganize your class and put variable definitions before methods (for clarity) (like `private ...` and `constructor`

File: sitn_portal/sitn-portal-client/src/app/docs/document.service.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    return this.http.get(this.documentsUrl)
+       .toPromise()
+       .then(response => response.json().documents as Document[])
+       .catch(this.handleError);
+     }
+
+  private headers = new Headers({'Content-Type': 'application/json'});
+
+  private handleError(error: any): Promise<any> {
+    console.error('An error occurred', error); // for demo purposes only
```
Comment (kalbermattenm) [inline sitn_portal/sitn-portal-client/src/app/docs/document.service.ts]: No consoles please.

File: sitn_portal/sitn-portal-client/src/app/docs/document-search.service.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { Document }           from './document';
+
+@Injectable()
+export class DocumentSearchService {
+
+  constructor(private http: Http) {}
+
+  search(term: string): Observable<Document[]> {
+    return this.http
+               .get(`documents/officialtitle=${term}`)
```
Comment (kalbermattenm) [inline sitn_portal/sitn-portal-client/src/app/docs/document-search.service.ts]: This is not good. You cannot use URLs like that, you have to use complete URLs, not relative ones. This comes in addition to my remark concerning Python routes.

File: sitn_portal/sitn-portal-client/src/index.html [hunk 5]
```diff
@@ -5,8 +5,14 @@
...
 
-  <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link rel="icon" type="image/x-icon" href="favicon.ico">
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<link rel="icon" type="image/x-icon" href="favicon.ico">
+
+  <link rel="stylesheet" href="styles.css">
+  <link rel="stylesheet" href="forms.css">
+  <link rel="stylesheet"
+        href="https://unpkg.com/bootstrap@3.3.7/dist/css/bootstrap.min.css">
```
Comment (kalbermattenm) [inline sitn_portal/sitn-portal-client/src/index.html]: What is this doing here ? We do not want any external http requested library. If you need bootstrap, then you should use it using npm.

Comment (kalbermattenm) [inline sitn_portal/sitn-portal-client/src/index.html]: Moreover, why in this file ? Did you change anything to `index.html`

File: sitn_portal/views/docs.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (kalbermattenm) [inline sitn_portal/views/docs.py]: My fault, sorry...

I did not review any thing else in this file...

File: sitn_portal/sitn-portal-client/src/app/docs/document.service.ts (hunk 1 above)
Comment (voisardf) [inline sitn_portal/sitn-portal-client/src/app/docs/document.service.ts]: don't know... I remove it and we'll see

File: sitn_portal/sitn-portal-client/src/index.html (hunk 5 above)
Comment (voisardf) [inline sitn_portal/sitn-portal-client/src/index.html]: I just added the css (forms)

File: sitn_portal/sitn-portal-client/src/app/docs/document.service.ts (hunk 1 above)
Comment (kalbermattenm) [inline sitn_portal/sitn-portal-client/src/app/docs/document.service.ts]: But you should test it before definitively removing it...

**Comment by voisardf** [comment]
merging as discussed, thanks

**Pull request merged**